esrally/metrics.py

```diff
@@ -254,8 +254,10 @@ def results_template(self):
     def _read(self, template_name):
         with open("%s/resources/%s.json" % (self.script_dir, template_name), encoding="utf-8") as f:
             template = json.load(f)
-            template["settings"]["index"]["number_of_shards"] = self._number_of_shards
-            template["settings"]["index"]["number_of_replicas"] = self._number_of_replicas
+            if self._number_of_shards is not None:
+                template["settings"]["index"]["number_of_shards"] = self._number_of_shards
+            if self._number_of_replicas is not None:
```

Comment:
Very good that you didn't `if self._number_of_shards` here (and above) because explicitly setting e.g. number of replicas to `0` would be evaluated as false and not set the value!